crypto/keys/secp256r1/pubkey.go

```diff
@@ -0,0 +1,59 @@
+package secp256r1
+
+import (
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+
+	ecdsa "github.com/cosmos/cosmos-sdk/crypto/keys/internal/ecdsa"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+)
+
+// String implements proto.Message interface.
+func (m *PubKey) String() string {
+	return m.Key.String(name)
+}
+
+// Bytes implements SDK PubKey interface.
+func (m *PubKey) Bytes() []byte {
+	return m.Key.Bytes()
+}
+
+// Equals implements SDK PubKey interface.
+func (m *PubKey) Equals(other cryptotypes.PubKey) bool {
+	pk2, ok := other.(*PubKey)
+	if !ok {
+		return false
+	}
+	return m.Key.Equal(&pk2.Key.PublicKey)
+}
+
+// Address implements SDK PubKey interface.
+func (m *PubKey) Address() tmcrypto.Address {
+	return m.Key.Address(m.XXX_MessageName())
```

Comment:
Using `proto.MessageName(m)` might read clearer

Reply:
This is how the MessageName is implemented - if you someone want's to customize the protobuf message name, (s)he will need to implement that method. Is it worth to add additional round-trip?

Reply:
Yes, imo we should never call `XXX_` methods in non-generated code. Plus, ADR-028 use proto.MessageName

Reply:
I intentionally added an option to  generate that function (both for this use case and `proto.MessageName`.  In my opinion it's fine to use it here - because we are in the same package and we know that this function is there (which is not always a case in other packages). I agree, that calling XXX method from external package should be avoided if possible.

Reply:
`proto.MessageName(m)` is more correct IMHO. Doesn't this get cached anyway?

Reply:
`proto.MessageName` firstly tries to call `XXX_MessageName`. If this interface is not implemented then it goes into reflect and a global cache (cache is using reflect type).

Reply:
Updated to use `proto.MessageName`